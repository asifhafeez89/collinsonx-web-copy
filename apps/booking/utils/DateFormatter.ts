import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import { DATE_TIME_FORMAT } from 'config/Constants';
import 'dayjs/locale/es';
dayjs.extend(utc);
dayjs.extend(timezone);

export function formatDate(
  dateval: Date,
  formattype: string,
  locale?: string
): string {
  dayjs.locale(locale ? locale : 'en');
  return dayjs(dateval).format(formattype);
}

export function formatTimezone(localTimeHour: string, timezone: string) {
  return dayjs.tz(localTimeHour, timezone).format(DATE_TIME_FORMAT);
}
