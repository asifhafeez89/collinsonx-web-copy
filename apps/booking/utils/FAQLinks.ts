export const FAQLink = (accountProvider?: string, locale?: string) => {
  const loungeKey_link_faq = `https://portal.loungekey.com/${locale}/elo/faq/`;
  const priorityPass_link_faq = `https://memberhelp.prioritypass.com/${locale}/support/home`;
  return accountProvider === 'LOUNGE_KEY'
    ? loungeKey_link_faq
    : accountProvider === 'PRIORITY_PASS'
    ? priorityPass_link_faq
    : '';
};
