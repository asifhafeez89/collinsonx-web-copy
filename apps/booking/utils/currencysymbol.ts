export const currencyMap: Record<string, string> = {
  GBP: String.fromCharCode(163),
};

export const getCurrencySymbol = (currency: string) =>
  currencyMap[currency] || currency;
