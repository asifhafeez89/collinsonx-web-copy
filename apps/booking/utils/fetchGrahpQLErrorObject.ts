// @ts-ignore
function fetchGrahpQLErrorObject(response, key: string) {
  if (!response && typeof response !== 'object') {
    return null;
  }

  if (!response[key]) {
    return null;
  }

  const error = response[key];

  if (error && 'graphQLErrors' in error) {
    const graphQLErrors = error.graphQLErrors;

    if (graphQLErrors && typeof graphQLErrors === 'object') {
      const data = error.graphQLErrors[0];
      if ('extensions' in data) {
        return data.extensions;
      }
    }
  }

  return null;
}

export default fetchGrahpQLErrorObject;
