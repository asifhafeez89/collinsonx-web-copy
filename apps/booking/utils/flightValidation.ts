const FLIGHT_VALIDATION = /([0-9]{0,1}[a-zA-Z]{1,3})([0-9]*)/gi;

export function validateFlightNumber(
  flight: string
): [boolean, string | undefined, string | undefined] {
  const match = [...flight.matchAll(FLIGHT_VALIDATION)];

  if (!match) {
    return [false, undefined, undefined];
  }

  const carrierLen = match[0]?.[1]?.length ?? 0;
  if (carrierLen < 1 || carrierLen > 3) {
    return [false, undefined, undefined];
  }

  const flightNoLen = match[0]?.[2]?.length ?? 0;
  if (flightNoLen < 1 || flightNoLen > 5) {
    return [false, undefined, undefined];
  }

  return [true, match[0][1], match[0][2]];
}
