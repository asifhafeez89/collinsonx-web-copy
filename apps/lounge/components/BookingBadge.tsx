import {
  Badge as MBadge,
  BadgeProps as MBadgeProps,
  Box,
  createStyles,
} from '@collinsonx/design-system/core';

export type BookingStatus = 'pending' | 'declined' | 'confirmed';

export interface BadgeProps extends MBadgeProps {
  status: 'pending' | 'confirmed' | 'declined';
}

const useStyles = createStyles(({ colors }) => {
  return {
    root: {
      borderRadius: 8,
      textTransform: 'none',
      fontSize: 16,
      height: 'auto',
      fontWeight: 600,
    },
    inner: {
      padding: '8px 16px',
    },
    declined: {
      backgroundColor: colors.red[7],
      color: '#FFF',
    },
    pending: {
      backgroundColor: colors.yellow[6],
      color: '#000',
    },
    confirmed: {
      backgroundColor: colors.cyan[6],
      color: '#000',
    },
  };
});

export default function BookingBadge({
  children,
  status,
  ...props
}: BadgeProps) {
  const { classes, cx } = useStyles();

  return (
    <MBadge
      {...props}
      className={cx(classes.root, classes.inner, {
        [classes.declined]: status === 'declined',
        [classes.pending]: status === 'pending',
        [classes.confirmed]: status === 'confirmed',
      })}
    >
      <Box>{children}</Box>
    </MBadge>
  );
}
