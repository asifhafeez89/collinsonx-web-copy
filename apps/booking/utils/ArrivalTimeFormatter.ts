import { formatDate } from './DateFormatter';
import { TIME_FORMAT } from 'config/Constants';

export const arrivalTimeFormatter = (
  bookedFrom: string | undefined,
  lastArrival: string | undefined
) =>
  `${formatDate(new Date(`${bookedFrom}`), TIME_FORMAT)}
  -
  ${formatDate(new Date(`${lastArrival}`), TIME_FORMAT)}`;
