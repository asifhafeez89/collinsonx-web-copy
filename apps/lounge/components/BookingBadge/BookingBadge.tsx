import {
  Badge as MBadge,
  BadgeProps as MBadgeProps,
  Box,
  Group,
  Flex,
  Text,
  createStyles,
} from '@collinsonx/design-system/core';
import styled from '@collinsonx/design-system/styled';

import {
  Declined as DeclinedIcon,
  Confirmed as ConfirmedIcon,
  Pending as PendingIcon,
} from '@collinsonx/design-system/assets/icons';

import { BookingStatus } from '@collinsonx/utils';
import {
  Confirmed as BookingConfirmed,
  Declined as BookingDeclined,
  Pending as BookingPending,
} from '@collinsonx/design-system/assets/icons';
import { start } from 'repl';

const {
  Initialized,
  Confirmed,
  CheckedIn,
  Declined,
  Cancelled,
  Booked,
  Errored,
} = BookingStatus;

export interface BadgeProps extends MBadgeProps {
  status: BookingStatus;
  largeBadge?: boolean;
}

const useStyles = createStyles(({ colors }) => {
  return {
    root: {
      borderRadius: 4,
      textTransform: 'none',
      fontSize: 14,
      lineHeight: 24,
      fontWeight: 400,
      color: '#25262B',
    },
    inner: {
      padding: '12px 8px',
    },
    declined: {
      backgroundColor: '#FCD8D8',
      border: '2px solid #F03E3E',
    },
    pending: {
      backgroundColor: '#FEEFCD',
      border: '2px solid #FAB005',
    },
    confirmed: {
      backgroundColor: '#E3F1D0',
      border: '2px solid #74B816',
      overflow: 'visible',
    },
    largeBadge: {
      alignSelf: 'flex-start',
    },
    smallBadge: {
      borderRadius: 4,
      outline: '4px solid #fff',
      width: 24,
      height: 24,
      padding: 0,
      display: 'block',
    },
  };
});

export const bookingMap: Record<BookingStatus, string> = {
  [Initialized]: 'Pending',
  [Confirmed]: 'Confirmed',
  [Declined]: 'Declined',
  [Cancelled]: 'Cancelled',
  [CheckedIn]: 'Booking checked in',
  [Booked]: 'Booking',
  [Errored]: 'Booking errored',
};

export default function BookingBadge({
  children,
  status,
  largeBadge,
  ...props
}: BadgeProps) {
  const { classes, cx } = useStyles();

  return (
    <MBadge
      {...props}
      className={cx(
        classes.root,
        classes.inner,
        {
          [classes.declined]:
            status === Declined || status === Cancelled || status === Errored,
          [classes.pending]: status === Initialized,
          [classes.confirmed]: status === Confirmed || status === CheckedIn,
        },
        largeBadge ? classes.largeBadge : classes.smallBadge
      )}
    >
      {largeBadge && <Group>{bookingMap[status]}</Group>}
      {!largeBadge && <h1>asd</h1>}
      {!largeBadge && status === 'DECLINED' && (
        <DeclinedIcon
          style={{
            top: '4px',
            left: '3px',
            position: 'absolute',
            color: '#C92A2A',
          }}
        />
      )}
      {!largeBadge && status === 'CONFIRMED' && (
        <ConfirmedIcon
          style={{
            top: '4px',
            left: '3px',
            position: 'absolute',
            color: '#456E0E',
          }}
        />
      )}
      {!largeBadge && status === BookingStatus.Initialized && (
        <PendingIcon
          style={{
            top: '4px',
            left: '3px',
            position: 'absolute',
            color: '#966A03',
          }}
        />
      )}
    </MBadge>
  );
}
