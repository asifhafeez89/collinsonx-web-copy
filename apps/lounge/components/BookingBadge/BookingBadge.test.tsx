import React from 'react';
import BookingBadge, { BadgeProps } from './BookingBadge';
import { BookingStatus } from '@collinsonx/utils';
import { bookingMap } from './BookingBadge';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

let badgeProps: BadgeProps;

describe('BookingCard component test', () => {
  beforeEach(() => {
    badgeProps = {
      status: BookingStatus.Confirmed,
      largeBadge: true,
    };
  });

  it('large badge renders correctly', () => {
    const { container } = render(<BookingBadge {...badgeProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('checks if status text appears in the large badge', () => {
    render(<BookingBadge {...badgeProps} />);
    expect(screen.getByText(bookingMap[badgeProps.status])).toBeInTheDocument();
  });
});
