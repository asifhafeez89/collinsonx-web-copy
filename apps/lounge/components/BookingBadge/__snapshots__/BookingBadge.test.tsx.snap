// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCard component test large badge renders correctly 1`] = `
.emotion-1 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-size: 0.6875rem;
  height: 1.25rem;
  line-height: calc(1.25rem - 0.125rem);
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0 calc(1rem / 1.5);
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: auto;
  text-transform: uppercase;
  border-radius: 2rem;
  font-weight: 700;
  letter-spacing: 0.015625rem;
  cursor: inherit;
  text-overflow: ellipsis;
  overflow: hidden;
  background: rgba(231, 245, 255, 1);
  color: #228be6;
  border: 0.0625rem solid transparent;
  border-radius: 4px;
  text-transform: none;
  font-size: 14px;
  line-height: 24;
  font-weight: 400;
  color: #25262B;
  padding: 12px 8px;
  background-color: #E3F1D0;
  border: 2px solid #74B816;
  overflow: visible;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
}

.emotion-1:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 1rem;
}

.emotion-5>* {
  box-sizing: border-box;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

<div
  class="emotion-0 emotion-1"
>
  <span
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      Confirmed
    </div>
  </span>
</div>
`;
